reyna/src/com/b2msolutions/reyna/Dispatcher.java

```diff
@@ -52,30 +58,74 @@ protected Result sendMessage(Message message, HttpPost httpPost, HttpClient http
     public static Result canSend(Context context) {
         ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
         NetworkInfo info = connectivityManager.getActiveNetworkInfo();
-        if(info == null || !info.isConnectedOrConnecting()) {
+        if (info == null || !info.isConnectedOrConnecting()) {
             return Result.NOTCONNECTED;
         }
 
-        TimeRange range = new Preferences(context).getCellularDataBlackout();
-        if (range == null) {
+        Preferences preferences = new Preferences(context);
+        BlackoutTime blackoutTime = new BlackoutTime();
+
+        if (TextUtils.isEmpty(preferences.getWwanBlackout())) {
+            saveCellularDataAsWwanForBackwardCompatibility(preferences);
+        }
+
+        if (Dispatcher.isBatteryCharging(context) && !preferences.canSendOnCharge()) return Result.BLACKOUT;
+        if (!Dispatcher.isBatteryCharging(context) && !preferences.canSendOffCharge()) return Result.BLACKOUT;
+        if (isRoaming(info) && !preferences.canSendOnRoaming()) return Result.BLACKOUT;
+        try {
+            if (isWifi(info) && !canSendNow(blackoutTime, preferences.getWlanBlackout())) return Result.BLACKOUT;
+            if (isMobile(info) && !canSendNow(blackoutTime, preferences.getWwanBlackout())) return Result.BLACKOUT;
+        } catch (ParseException e) {
```

Comment:
Is it better to catch any exception instead of just ParseException ?
@Youhana-Hana